libraries/AP_SerialManager/AP_SerialManager.cpp

```diff
@@ -41,10 +41,14 @@ extern const AP_HAL::HAL& hal;
 #endif
 
 #ifndef HAL_SERIAL6_PROTOCOL
-#define SERIAL6_PROTOCOL SerialProtocol_None
-#define SERIAL6_BAUD AP_SERIALMANAGER_MAVLINK_BAUD/1000
```

Comment:
we should only default a port to SLCAN if we have CAN support and OTG2 is used in the hwdef.dat. I think the hwdef.dat parser could set the SERIALn_PROTOCOL correctly for SLCAN when it sees OTG2 in the list.